$OpenBSD$

SPDX-License-Identifier: AGPL-3.0-only

Send service certificates in the SMP handshake rather than as TLS client
credentials, avoiding TLS client-auth EKU restrictions for service certs.

Index: src/Simplex/Messaging/Client.hs
--- src/Simplex/Messaging/Client.hs.orig
+++ src/Simplex/Messaging/Client.hs
@@ -600,7 +606,7 @@
     runClient :: (ServiceName, ATransport 'TClient) -> TransportHost -> PClient v err msg -> IO (Either (ProtocolClientError err) (ProtocolClient v err msg))
     runClient (port', ATransport t) useHost c = do
       cVar <- newEmptyTMVarIO
-      let tcConfig = (transportClientConfig networkConfig nm useHost useSNI useALPN) {clientCredentials = serviceCreds <$> serviceCredentials}
+      let tcConfig = transportClientConfig networkConfig nm useHost useSNI useALPN
           socksCreds = clientSocksCredentials networkConfig proxySessTs transportSession
       tId <-
         runTransportClient tcConfig socksCreds useHost port' (Just $ keyHash srv) (client t c cVar)

